test/elm/interval/data.cql

```diff
@@ -51,47 +51,55 @@ define ContainsQuantity: Interval[1 'mg', 5 'mg'] contains 3 'mg'
+define NotContainsImpreciseDate: DateIvlHighClosed contains DateTime(2012, 10)
+define MayContainImpreciseDate: DateIvlHighOpen contains DateTime(2012)
+define PrecisionDateIvlHighOpen: Interval[DateTime(2012, 3, 2, 12, 34, 56, 789), DateTime(2012, 9, 2, 1, 23, 45, 678))
+define PrecisionDateIvlHighClosed: Interval[DateTime(2012, 3, 2, 12, 34, 56, 789), DateTime(2012, 9, 2, 1, 23, 45, 678)]
+define ContainsDayOfDateLowEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3, 2, 0, 0, 0, 0)
+define NotContainsDayOfDateHighEdgeOpen: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9, 2, 23, 59, 59, 999)
+define ContainsDayOfDateHighEdgeClosed: PrecisionDateIvlHighClosed contains day of DateTime(2012, 9, 2, 23, 59, 59, 999)
+define NotContainsDayOfDateLowEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3, 1, 23, 59, 59, 999)
+define NotContainsDayOfDateBeyondHighEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9, 3, 0, 0, 0, 0)
+define ContainsDayOfDateImpreciseLowEdge: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3, 2)
+define NotContainsDayOfDateImpreciseHighEdgeOpen: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9, 2)
+define ContainsDayOfDateImpreciseHighEdgeClosed: PrecisionDateIvlHighClosed contains day of DateTime(2012, 9, 2)
+define ContainsDayOfDateVeryImpreciseMiddle: PrecisionDateIvlHighOpen contains day of DateTime(2012, 6)
+define NotContainsDayOfDateVeryImpreciseLow: PrecisionDateIvlHighOpen contains day of DateTime(2012, 2)
+define NotContainsDayOfDateVeryImpreciseHighEdgeOpen: PrecisionDateIvlHighOpen contains day of DateTime(2012, 10)
+define MayContainDayOfDateVeryImpreciseLow: PrecisionDateIvlHighOpen contains day of DateTime(2012, 3)
+define MayContainDayOfDateVeryImpreciseHigh: PrecisionDateIvlHighOpen contains day of DateTime(2012, 9)
+define MayContainDayOfDateVeryImpreciseSurrounding: PrecisionDateIvlHighOpen contains day of DateTime(2012)
 define ImpDateIvl: Interval[DateTime(2012, 3), DateTime(2012, 8)]
 define ImpreciseContainsDate: ImpDateIvl contains DateTime(2012, 6, 1, 0, 0, 0, 0)
 define ImpreciseNotContainsDate: ImpDateIvl contains DateTime(2012, 9, 1, 0, 0, 0, 0)
 define ImpreciseMayContainDate: ImpDateIvl contains DateTime(2012, 3, 15, 0, 0, 0, 0)
 define NegInfBegContainsInt: Interval[null, 5] contains -7
 define NegInfBegNotContainsInt: Interval[null, 5] contains 7
-define UnknownBegContainsInt: Interval(null, 5] contains 5
+define unknownOpenBegMayContainInt: Interval(null, 5] contains 5
+define unknownClosedBegContainsInt: Interval[null, 5] contains 5
 define UnknownBegMayContainInt: Interval(null, 5] contains -7
 define UnknownBegNotContainsInt: Interval(null, 5] contains 7
 define PosInfEndContainsInt: Interval[0, null] contains 123456789
 define PosInfEndNotContainsInt: Interval[0, null] contains -1
-define UnknownEndContainsInt: Interval[0, null) contains 0
+define UnknownOpenEndMayContainInt: Interval[0, null) contains 0
```

Comment:
And another CQL test that I think should result in `true`, but doesn't according to current definition of `contains`.